### ISSUE TITLE
Clarifications on start and the run the backend server

### ISSUE THREAD

**Issue by sabahtalateh** [saschagrunert/webapp.rs#573]
Hi @saschagrunert ) Can you please clarify the place where you start the server in `from_config ` method https://github.com/saschagrunert/webapp.rs/blob/master/backend/src/server/mod.rs#L86, and then start system runner in `start` method https://github.com/saschagrunert/webapp.rs/blob/master/backend/src/server/mod.rs#L102? What is the purpose of it? As I understand the server will start without calling the `start` method https://github.com/saschagrunert/webapp.rs/blob/master/backend/src/main.rs#L43, at least on my machine it works like that. If I don't start the server in `from_config` then I should start by calling `start` which in turn calls `runner.run`.. 
So to sum up. What is the purpose of starting server in `from_config` and the running system runner in `start`?
Thanks

**Comment by saschagrunert** [comment]
Hey @sabahtalateh, I think it would be possible to combine them into one method. That the start is not needed anymore seems a somewhat new feature of actix-web, so we could merge them into one. :)

**Comment by saschagrunert** [comment]
But in the end it has been added to be unit-testable without the reason to actually  run the actix system.

**Comment by sabahtalateh** [comment]
@sabahtalateh Oh.. I was wrong thinking that `server.start()` actually starts the server, it is a builder method that tells that the server should be started when the system run
